trino/dbapi.py

```diff
@@ -454,6 +454,20 @@ def execute(self, operation, params=None):
                 self._deallocate_prepared_statement(statement_name)
 
         else:
+            if params:
+                assert isinstance(params, (list, tuple)), (
+                    'params must be a list or tuple containing the query '
+                    'parameter values'
+                )
+
+                # substitue parameters in query in reversed order
+                question_mark_positions = [index for index, character in enumerate(operation) if character == '?']
```

Comment:
Alternative to the reverse trick is to use string slicing. maybe that would be slightly more readable?